### ISSUE TITLE
Files.app requests all files within a folder which leads to hangs when the folder is large

### ISSUE THREAD

**Issue by dblsaiko** [blinksh/blink#2064]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

Blink 18.0.1, iOS 18.0.1

### Describe the bug

In certain situations Files.app tries to recursively access every file in a folder. I don't know why, might be to prepare thumbnails. The Blink file provider will then actually transfer each file completely to the phone, which takes a very long time if there are large files or many files in the folder. This prevents performing pretty much any action in Files.app until it is finished, for example viewing or sharing a file.

Here's a situation in which I've noticed this happen:

- ~ (Remote folder added to Blink)
- ~/Desktop/ (navigated here)
- ~/Desktop/folder
- ~/Desktop/folder/largefile.iso (10 GB)
- ~/Desktop/screenshot.png (1 MB, the file I want to share)

In this situation, trying to select screenshot.png in Files.app highlights the file but does not open the preview, similarly sharing from the context menu does nothing. Afterwards it is no longer possible to open the context menu either. Looking at the network monitor it is continuously transferring data in the background.

Moving Desktop/folder somewhere else keeps it from being accessed and I can interact normally.

I've also had this happen before when Desktop/folder instead contains more folders with many smaller files inside but also several GB in total.

Might be fixed with #1717.

### Expected behavior
Large folders are handled in a performant way.

**Comment by carloscabanero** [comment]
Yeah, so I don't think the issue is that things are getting downloaded by the extension, but the behavior in iOS18 has changed and as other users also noticed, things are hanging up for apparently no reason. This may also happen when switching to a different section in the Files.app after browsing down the hierarchy.

As you already mentioned, we are moving to #1717, but let's keep this open here and referenced so we can make sure the new one is behaving properly. From my tests, I know it already does.

Contact me if you are not on Community (TestFlight) yet, so you can test once I release 18.1.0. Thanks!

**Comment by dblsaiko** [comment]
Hm, I feel like this was already happening before iOS 18 but I might be wrong. I was using the beta for a while so I don't remember the first time I saw this exactly.

> Contact me if you are not on Community (TestFlight) yet, so you can test once I release 18.1.0. Thanks!

Alright, I sent an email.

**Comment by carloscabanero** [comment]
Closing. Solved on new provider.
